aerleon/lib/naming.py

```diff
@@ -538,6 +542,29 @@ def GetServiceByProto(self, query: str, proto: str) -> List[str]:
                     services_set.add(parts[0])
         return sorted(services_set)
 
+    def GetFQDN(self, query: str) -> List[str]:
+        returnlist = []
+        data = query.split('#')
+        token = data[0].split()[0]
+        if token not in self.fqdn:
+            raise UndefinedAddressError(f'UNDEFINED: {token}')
+        for i in self.fqdn[token].items:
+            comment = ''
+            if i.find('#') > -1:
+                (name, comment) = i.split('#', 1)
+            else:
+                name = i
+
+            name = name.strip()
+            # Check if hostname is actually a subtoken
+            if self.token_re.match(name):
+                returnlist.extend(self.GetFQDN(name))
+            elif self.fqdn_re.match(name):
+                returnlist.append(FQDN(name, token, comment))
+            else:
+                logging.debug(f"{name} did not match FQDN")
```

Comment:
In GetNet, we have the IP constructor raise an error if the input is not a valid IP address. Can we move the FQDN validation logic into the FQDN constructor to mirror how we use the IP constructor? Then, `fqdn_re` can be encapsulated as it is a concern of the FQDN constructor and not the naming library.

Reply:
They should be the same now.